api_types.ts

```diff
@@ -1107,6 +1139,99 @@ export enum PrecannedDateRange {
   Everything = 'everything',
 }
 
+/**
+ * The set of date ranges whose end dates are today.
+ */
+export const PastRelativeDateRanges = [
+  PrecannedDateRange.Today,
+  PrecannedDateRange.Last7Days,
+  PrecannedDateRange.Last30Days,
+  PrecannedDateRange.Last90Days,
+  PrecannedDateRange.Last180Days,
+  PrecannedDateRange.Last365Days,
+  PrecannedDateRange.YearToDate,
+];
+
+/**
+ * The set of date ranges that are useful for filtering datasets that don't include
+ * future dates.
+ */
+export const PastDateRanges = [
+  ...PastRelativeDateRanges,
+  PrecannedDateRange.Yesterday,
+  PrecannedDateRange.LastWeek,
+  PrecannedDateRange.LastMonth,
+  PrecannedDateRange.LastYear,
+  PrecannedDateRange.ThisWeek,
+  PrecannedDateRange.ThisMonth,
+  PrecannedDateRange.ThisYear,
+  PrecannedDateRange.Everything,
+];
+
+/**
+ * The set of date ranges whose start dates are today.
+ */
+export const FutureRelativeDateRanges = [
+  PrecannedDateRange.Today,
+  PrecannedDateRange.Next7Days,
+  PrecannedDateRange.Next30Days,
+  PrecannedDateRange.Next90Days,
+  PrecannedDateRange.Next180Days,
+  PrecannedDateRange.Next365Days,
+];
+
+/**
+ * Some APIs require relative dates only. Before we supported {@link ParamDef.allowedPrecannedValues},
+ * some packs decided to use a Date parameter for an input like this, because not all
+ * PrecannedDateRange values were valid.
+ *
+ * We want such packs to be able to use relative date ranges without needing to change their
+ * parameter type, to maintain backwards compatibility.
+ */
+export enum PrecannedDate {
+  Today = 'today',
+  Yesterday = 'yesterday',
+  Tomorrow = 'tomorrow',
+  DaysAgo7 = '7 days ago',
```

Comment:
Any reason to put spaces in these? It doesn't really matter but we have underscores in the existing enums, which may or may not be to match similar `coda` enums. Just trying to have some consistency in the shape of these values.

Reply:
changed to underscores